The parameter rest_total_hits_as_int in HLRC in the next scroll request iteration is missing

**Elasticsearch version** (`bin/elasticsearch --version`): 7.8.0

**HLRC version ** : 6.8.12

**Plugins installed**: []

**JVM version** (`java -version`):   "1.8.0_171"

**OS version** (`uname -a` if on a Unix-like system): macOs Catalina 10.15.4

**Description of the problem including expected versus actual behavior**:
The scroll search request has the parameter rest_total_hits_as_int ([#46076](https://github.com/elastic/elasticsearch/pull/46076)) but in the next scroll iterations the parameter is missing.

**Steps to reproduce**:

 1.  Start a 7.8.0 ES cluster
 2. Using the version 6.8 HLRC  in your code
 <dependency>
            <groupId>org.elasticsearch</groupId>
            <artifactId>elasticsearch</artifactId>
            <version>6.8.4</version>
  </dependency>

 3. Write the next test:

```
import com.google.common.collect.ImmutableMap;
import org.apache.http.HttpHost;
import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest;
import org.elasticsearch.action.admin.indices.refresh.RefreshRequest;
import org.elasticsearch.action.index.IndexRequest;
import org.elasticsearch.action.search.SearchRequest;
import org.elasticsearch.action.search.SearchResponse;
import org.elasticsearch.action.search.SearchScrollRequest;
import org.elasticsearch.client.RequestOptions;
import org.elasticsearch.client.RestClient;
import org.elasticsearch.client.RestHighLevelClient;
import org.elasticsearch.common.unit.TimeValue;
import org.elasticsearch.index.query.QueryBuilders;
import org.elasticsearch.search.builder.SearchSourceBuilder;
import org.junit.Test;

@Test
    public void testTotalHitsInScrollSearch() throws IOException {
        String indexName = "users";
        RestHighLevelClient client = new RestHighLevelClient(RestClient.builder(new HttpHost("localhost", 9200, "http")));

        client.indices().delete(new DeleteIndexRequest(indexName), RequestOptions.DEFAULT);

        for (int i = 0; i < 100; i++) {
            Map<String, Object> json = ImmutableMap.of("id", i, "status", "active");
            client.index(new IndexRequest(indexName, indexName).source(json), RequestOptions.DEFAULT);
        }
        client.indices().refresh(new RefreshRequest(indexName), RequestOptions.DEFAULT);

        SearchRequest request = new SearchRequest(indexName);
        SearchSourceBuilder source = new SearchSourceBuilder();
        source.size(5);
        source.query(QueryBuilders.termQuery("status", "active"));
        request.scroll(TimeValue.timeValueMinutes(1));
        request.source(source);
        SearchResponse response = client.search(request, RequestOptions.DEFAULT);
        assertEquals(5, response.getHits().getHits().length);
        assertEquals(100, response.getHits().getTotalHits());

        SearchScrollRequest request2 = new SearchScrollRequest(response.getScrollId());
        response = client.scroll(request2, RequestOptions.DEFAULT);
        assertEquals(5, response.getHits().getHits().length);
        assertEquals(100, response.getHits().getTotalHits());

    }
```

**Provide logs (if relevant)**:  Test output
```
java.lang.AssertionError: 
Expected :100
Actual   :0
```

--- a/client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java
+++ b/client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java
@@ -409,8 +409,14 @@ final class RequestConverters {
         }
     }
 
+    private static void addSearchScrollRequestParams(Params params) {
+        params.putParam(RestSearchAction.TOTAL_HIT_AS_INT_PARAM, "true");
+    }
+
     static Request searchScroll(SearchScrollRequest searchScrollRequest) throws IOException {
         Request request = new Request(HttpPost.METHOD_NAME, "/_search/scroll");
+        Params params = new Params(request);
+        addSearchScrollRequestParams(params);
         request.setEntity(createEntity(searchScrollRequest, REQUEST_BODY_CONTENT_TYPE));
         return request;
     }

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java
@@ -1150,10 +1150,14 @@ public class RequestConvertersTests extends ESTestCase {
         if (randomBoolean()) {
             searchScrollRequest.scroll(randomPositiveTimeValue());
         }
+        Map<String, String> expectedParams = new HashMap<>();
+        expectedParams.put(RestSearchAction.TOTAL_HIT_AS_INT_PARAM, "true");
+
         Request request = RequestConverters.searchScroll(searchScrollRequest);
         assertEquals(HttpPost.METHOD_NAME, request.getMethod());
         assertEquals("/_search/scroll", request.getEndpoint());
-        assertEquals(0, request.getParameters().size());
+        assertEquals(1, request.getParameters().size());
+        assertEquals(expectedParams, request.getParameters());
         assertToXContentBody(searchScrollRequest, request.getEntity());
         assertEquals(REQUEST_BODY_CONTENT_TYPE.mediaTypeWithoutParameters(), request.getEntity().getContentType().getValue());
     }

